### ISSUE TITLE
Getting null for PULL_REQUEST_ID and PULL_REQUEST_URL

### ISSUE THREAD

**Issue by RishabhTayal** [tomasbjerre/pull-request-notifier-for-bitbucket#175]
Below is my setup on my stash server(bitbucket server) and Jenkins.

**Jenkins Groovy Command:**
```
def currentBuild = Thread.currentThread().executable
def PULL_REQUEST_URL = build.buildVariableResolver.resolve('PULL_REQUEST_URL')
def PULL_REQUEST_ID = build.buildVariableResolver.resolve('PULL_REQUEST_ID')
def description = "<a href='$PULL_REQUEST_URL'>PR #$PULL_REQUEST_ID</a>"
currentBuild.setDescription(description)
```

**Jenkins URL in Stash Pull request notification configuration:**
`http://<Jenkins_URL>/git/notifyCommit?url=${PULL_REQUEST_TO_HTTP_CLONE_URL}&branches=pr/${PULL_REQUEST_TO_BRANCH}&sha1=${PULL_REQUEST_FROM_HASH}&PULL_REQUEST_URL=${PULL_REQUEST_URL}&PULL_REQUEST_ID=${PULL_REQUEST_ID}`

**Screenshot of Jenkins:**
![screenshot 2016-12-27 11 39 48](https://cloud.githubusercontent.com/assets/2100166/21505025/393b9b30-cc29-11e6-952a-cc505aa76de2.png)

As you can see in the screenshot the `PULL_REQUEST_URL` and `PULL_REQUEST_ID` are shown as `null`. Am I missing anything or is this a bug?

**Comment by tomasbjerre** [comment]
Did you try inoking Jenkins with a curl command to check if that works? If that works it may be an issue with the plugin.

**Comment by RishabhTayal** [comment]
Thanks for the quick response. I haven't tried invoking Jenkins with curl command. I am not sure how to do that.

**Comment by RishabhTayal** [comment]
@tomasbjerre any updates?

**Comment by tomasbjerre** [comment]
There are some examples of using curl in the readme.

**Comment by RishabhTayal** [comment]
When I run the curl command to I get the expected response back, but build is not triggered on Jenkins this time.

**Comment by tomasbjerre** [comment]
What exactly is the curl command that you run?


Den 28 dec. 2016 6:32 PM skrev "Rishabh Tayal" <notifications@github.com>:

**Comment by RishabhTayal** [comment]
```
curl -u USERNAME:PASSWORD 'http://BITBUCKET_SERVER_URL/rest/prnfb-admin/1.0/settings/notifications' -H 'Content-Type: application/json; charset=UTF-8' -H 'Accept: application/json, text/javascript, */*; q=0.01' --data-binary '{"uuid":"23e6c4f0-63d0-46c6-b129-293351407f47","name":"Notification","projectKey":"","repositorySlug":"","filterString":"","filterRegexp":"","triggers":["COMMENTED"],"injectionUrl":"","injectionUrlRegexp":"","user":"","password":"","proxyUser":"","proxyPassword":"","proxyServer":"","proxyPort":"","url":"http://localhost:80/?abc","method":"GET","postContent":"","headers":[{"name":"","value":""}]}'
```

I also run this by replacing the `name`, `projectKey` and `repositorySlug` with my repository's values.

**Comment by tomasbjerre** [comment]
You should run curl against Jenkins. To verify your Jenkins setup is working.

**Comment by RishabhTayal** [comment]
Jenkins setup is working fine. The builds are triggered everytime there is a pull request. The only issue is that the `PULL_REQUEST_ID` and `PULL_REQUEST_URL` are shown as `null`.

**Comment by tomasbjerre** [comment]
And are you able to trigger Jenkins with curl and get the values set
correctly?

Den 28 dec. 2016 7:05 PM skrev "Rishabh Tayal" <notifications@github.com>:

**Comment by RishabhTayal** [comment]
Sorry to keep asking stupid questions, but how can I trigger Jenkins? What url should I use? Is it there in your README.MD?

**Comment by RishabhTayal** [comment]
@tomasbjerre I just used the button to trigger the pull request build on bit bucket server and monitor the request and response in the network tab in the browser.

Here is the response I got from Jenkins after clicking the button.
```
{
   "confirmation":"on",
   "notificationResponses":[
      {
         "content":"Scheduled MyJob\nNo Git consumers using SCM API plugin for: http://abcd@git.mydomain.com/scm/project_name/repo.git\n”,
         "notification":"10b4ac9d-3a27-46db-806f-xxxxxxxxx”,
         "notificationName":"Notification",
         "status":200,
         "uri":"http%3A%2F%2Fuser%40git.mydomain.com%3A7994%2Fscm%2Fproject_name%2Frepo.git%26branches%3Dpr%2Fdevelop%26sha1%3Dxxxxxxxxxxxx%26PULL_REQUEST_URL%3Dhttps%3A%2F%2Fgit.mydomain.com%2Fprojects%2Fproject_name%2Frepos%2Fmy_repo%2Fpull-requests%2F1156%26PULL_REQUEST_ID%3D1234”
      }
   ]
}
```

Do you think the issue is the `uri` is encoded?

**Comment by tomasbjerre** [comment]
No it looks correct. The variables are resolved there in the URL. So the
issue is most likely with your Jenkins setup.

Den 28 dec. 2016 9:53 PM skrev "Rishabh Tayal" <notifications@github.com>:

**Comment by RishabhTayal** [comment]
Here are the screenshots of the Jenkins setup.

<details><summary>Screenshot 1</summary>
<img width="1005" alt="screenshot 2016-12-28 15 25 16" src="https://cloud.githubusercontent.com/assets/2100166/21532096/9620769c-cd12-11e6-9fe0-dd2655d8e87c.png">
</details>

<details><summary>Screenshot 2</summary>

![screenshot 2016-12-28 15 29 52](https://cloud.githubusercontent.com/assets/2100166/21532098/99c91bb4-cd12-11e6-8aac-b5417dc7b55d.png)
</details>

**Comment by tomasbjerre** [comment]
Follow the instructions in the readme

**Comment by tomasbjerre** [comment]
You are nissing the parameters.

**Comment by RishabhTayal** [comment]
really hard to follow. Not able to figure out where the issue is. Little help will be greatly appreciated.

**Comment by bingshiao** [comment]
@RishabhTayal select "This project is parameterized" in General tab and define parameters PULL_REQUEST_ID and PULL_REQUEST_URL explicitly

**Comment by RishabhTayal** [comment]
@bingshiao That doesn't work either. Now I see `<a href=''>PR #</a>` instead of `<a href='null'>PR #null</a>`

**Comment by RishabhTayal** [comment]
Can we reopen this issue? Still haven't find a solution for this.

**Comment by tomasbjerre** [comment]
I dont see any indication of it being a problem with this plugin.

**Comment by pranjalihande** [comment]
@RishabhTayal : I want to integrate jenkins with satsh in a way like whenever there is any pull request created/opened in stash it should start my jenkins job.

I have used similar setup like yours but job is not getting triggered when there is any pull request created from stash.

Can you please help me out here?

**Comment by pranjalihande** [comment]
I have used command as below 
 http://<jenkins URL>/jenkins/git/notifyCommit?url=${PULL_REQUEST_TO_SSH_CLONE_URL}&branches=pr/${PULL_REQUEST_TO_BRANCH}&sha1=${PULL_REQUEST_FROM_HASH}&PULL_REQUEST_URL=${PULL_REQUEST_URL}&PULL_REQUEST_ID=${PULL_REQUEST_ID} 

these parameters are added in this command. I have used in stash notifier

**Comment by RishabhTayal** [comment]
Correct but inorder to read these parameters you need to pass them in the jenkins command as well. I believe Jenkins changed their security configuration and now require you to paas the safe parameters when starting the Jenkins instance.
